### PR TITLE
feat: provide initial revolution number for Propagated model

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp [hunk 1]
```diff
@@ -44,22 +44,26 @@ class Propagated : public ostk::astro::trajectory::orbit::Model
     /// @brief Constructor
     ///
     /// @code{.cpp}
-    ///              Propagated propagated = { aPropagator, aState } ;
+    ///              Propagated propagated = { aPropagator, aState, aRevolutionNumber } ;
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp:47]:
```suggestion
    ///              Propagated propagated = { aPropagator, aState } ;
    /// 	     or
    ///              Propagated propagated = { aPropagator, aState, aRevolutionNumber } ;
```

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp [hunk 2]
```diff
@@ -44,22 +44,26 @@ class Propagated : public ostk::astro::trajectory::orbit::Model
...
     /// @param aState A state
-    Propagated(const Propagator& aPropagator, const State& aState);
+    /// @param aRevolutionNumber A revolution number
+    Propagated(const Propagator& aPropagator, const State& aState, const Integer& aRevolutionNumber = 1);
 
     /// @brief Constructor with a cached state array
     ///
     /// @code{.cpp}
-    ///              Propagated propagated = { aPropagator, aCachedStateArray } ;
+    ///              Propagated propagated = { aPropagator, aCachedStateArray, aRevolutionNumber } ;
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp:58]:
```suggestion
    ///              Propagated propagated = { aPropagator, aCachedStateArray } ;
    ///		     or
    ///              Propagated propagated = { aPropagator, aCachedStateArray, aRevolutionNumber } ;
```

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.cpp [hunk 3]
```diff
@@ -264,7 +268,7 @@ Integer Propagated::calculateRevolutionNumberAt(const Instant& anInstant) const
     Vector3d currentPositionCoordinates = currentPosition.inUnit(Position::Unit::Meter).accessCoordinates();
     Vector3d currentVelocityCoordinates = currentVelocity.inUnit(Velocity::Unit::MeterPerSecond).accessCoordinates();
 
-    // Determine whether to count revolution numbers in forwards or backwards time and return function if duration is 0
+    // Determine whether to count revolution numbers in forwards or backwards time and return function if duration is
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.cpp:271]: ? If duration is what?

**Review by apaletta3 — APPROVED** [review]
Looks good to me!

Should the default intialRev number specified actually be 0 instead of 1 though? Or does that go against the way we count rev nubmers in our services and the rest of OSTk?

**Review by Derollez — APPROVED** [review]

**Pull request merged**